plugins/bcli.c

```diff
@@ -595,8 +595,12 @@ static struct command_result *process_getrawblock(struct bitcoin_cli *bcli)
 	struct json_stream *response;
 	struct getrawblock_stash *stash = bcli->stash;
 
-	/* -1 to strip \n and steal onto the stash. */
-	bcli->output[bcli->output_bytes-1] = 0x00;
+	/* Strip trailing whitespace and steal onto the stash. */
+	size_t striplen = bcli->output_bytes;
+	while (isspace(bcli->output[striplen-1]))
```

Comment:
We have a cisspace helper which is sightly better here, BTW.

Can we extract this into a "static void strip_trailing_whitespace(char *str, size_t len)" 
which also makes sure it doesn't underflow if it's all whitespace?

Thanks!

Reply:
Done.